### PR TITLE
✨ Add first trigger, action and apps

### PR DESCRIPTION
**Pull request by lmachens**
The base for triggers, events and apps are set 🤘.
Now, we are able to work on #22.

The `triggerTarget` was a difficult one, especially because of the complex types. 
But I am very happy with the result. 

Tests are added.

### CODE REVIEW

File: src/triggers/leagueQueue.test.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          },
+        },
+      }
+    );
+    const handleLeagueQueue = jest.fn();
+
+    triggerTarget.addEventListener("leagueQueue", handleLeagueQueue);
+  });
+
+  it("clean up after event lisenter is removed", () => {
```
Comment (phlgr) [inline src/triggers/leagueQueue.test.ts:141]:
```suggestion
  it("cleans up after event listener is removed", () => {
```

File: src/triggers/triggerTarget.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/* no-non-null-assertion is disabled to allow access of this.listeners[name] */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { triggersObject } from ".";
+import type { TriggerTarget } from "./types";
+
+const triggerTarget: TriggerTarget = {
+  listeners: {},
+  dispatches: {},
```
Comment (phlgr) [inline src/triggers/triggerTarget.ts:9]: Either `listens` and `dispatches` or `listeners` and `dispatchers`

File: src/triggers/triggerTarget.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  dispatchEvent(name, data) {
+    if (!(name in this.listeners)) {
+      return;
+    }
+    const stack = this.listeners[name]!.slice();
+    for (let i = 0, l = stack.length; i < l; i++) {
+      const listener = stack[i] as (arg: typeof data) => void;
+      listener.call(this, data);
+    }
```
Comment (phlgr) [inline src/triggers/triggerTarget.ts:43]:
```suggestion
    stack.forEach((listener: (arg: typeof data) => void) => {
      listener.call(this, data);
    });
```

**Review by phlgr — CHANGES_REQUESTED** [review]
Will look into it further tomorrow 😄

File: src/triggers/triggerTarget.ts (hunk 3 above)
Comment (lmachens) [reply src/triggers/triggerTarget.ts:43]: 🤦‍♂️I shouldn't just copy & paste this example:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget#simple_implementation_of_eventtarget

**Review by phlgr — APPROVED** [review]

**Pull request merged**